jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -51,7 +52,9 @@ class AssignWcsStep(Step):
                             'fore', 'fpa', 'msa', 'ote', 'ifupost',
                             'ifufore', 'ifuslicer']
 
-    def process(self, input, *args, **kwargs):
+    def process(self, input, sip_approx=True, max_pix_error=0.25,
+                degree=None, max_inv_pix_error=0.25, inv_degree=None,
+                npoints=32, *args, **kwargs):
```

Comment:
@jdavies-st Shouldn't a step's parameters be specified through the `spec` "docstring"-like class member of the step? If this is not going to be modified ever, why even bother with this and not hardcode all these parameters?

Reply:
None of these parameters are documented. Shouldn't they be? Are these "private" parameters that users should not use? Then why have them? If they are supposed to be public - please add documentation.

Reply:
I'm probably responsible for that. Mea culpa.

Reply:
@mcara is right, all input parameters need to go in the spec section to be picked up by `configobj`.
@cshanahan1 Follow the examples in this file and other *_step.py files i fnecessary and add all new parameters to `spec`. They also need to be added to `jwst/pipeline/assign_wcs.cfg`.

Reply:
@mcara where should i add documentation, unless you were referring to adding documentation in gwcs? I wrote short descriptions of the parameters in ``spec``, let me know if this is sufficient!

Reply:
@nden would know better but my guess would be for something like https://github.com/spacetelescope/jwst/blob/master/docs/jwst/assign_wcs/main.rst. Take a look at https://github.com/spacetelescope/jwst/blob/master/docs/jwst/tweakreg/README.rst

Reply:
Yes, add information about the SIP approximation to the `Description` section of assign_wcs.